src/bodse/avl/adjust.py

```diff
@@ -0,0 +1,526 @@
+import pathlib
+import re
+import zipfile
+from typing import Optional
+
+# Third Party
+import numpy as np
+import pandas as pd
+from caf.toolkit import config_base, log_helpers
+from pydantic import dataclasses, types
+
+# Local Imports
+import bodse
+from bodse import utils
+from bodse.avl import database, gtfs
+
+##### CONSTANTS #####
+
+LOG = logging.getLogger(__name__)
+_GTFS_FILE_NAMES = {"stops": "stops.txt", "stop_times": "stop_times.txt"}
+_GTFS_FILE_COLUMNS: dict[str, dict[str, type]] = {
+    "stops": {
+        "stop_id": str,
+        "stop_code": str,
+        "stop_name": str,
+        "stop_lat": float,
+        "stop_lon": float,
+        "wheelchair_boarding": int,
+        "location_type": float,
+        "parent_station": str,
+        "platform_code": str,
+    },
+    "stop_times": {
+        "trip_id": str,
+        "arrival_time": str,
+        "departure_time": str,
+        "stop_id": str,
+        "stop_sequence": int,
+        "stop_headsign": str,
+        "pickup_type": int,
+        "drop_off_type": int,
+        "shape_dist_traveled": float,
+        "timepoint": int,
+        "stop_direction_name": str,
+    },
+}
+_STOP_TIMES_INDICATOR_COLUMN = "dataset"
+
+##### CLASSES & FUNCTIONS #####
+
+
+@dataclasses.dataclass(config={"arbitrary_types_allowed": True})
+class _GTFSStops:
+    """GTFS schedule stops and stop time data."""
+
+    stops: pd.DataFrame
+    stop_times: pd.DataFrame
+
+
+def _load_stop_times(path: pathlib.Path) -> _GTFSStops:
+    """Load GTFS stop locations and stop times tables.
+
+    Raises
+    ------
+    FileNotFoundError
+        If tables are missing from the GTFS file.
+    """
+    LOG.info("Loading stop times and locations from GTFS file: %s", path.name)
+    missing = []
+    data = {}
+
+    with zipfile.ZipFile(path) as gtfs_zip:
+        for name, filename in _GTFS_FILE_NAMES.items():
+            try:
+                with gtfs_zip.open(filename) as file:
+                    data[name] = pd.read_csv(file, dtype=_GTFS_FILE_COLUMNS[name])
+            except KeyError:
+                missing.append(name)
+
+    if len(missing) > 0:
+        raise FileNotFoundError(
+            f"{len(missing)} files missing from GTFS file ({path.name}): {missing}"
+        )
+
+    return _GTFSStops(data["stops"], data["stop_times"])
+
+
+def _translate_stop_coordinates(stops: pd.DataFrame) -> pd.DataFrame:
+    """Translate stops longitude and latitude coordinates to easting / northing.
+
+    Parameters
+    ----------
+    stops : pd.DataFrame
+        DataFrame containing stop locations, requires columns
+        'stop_lat' and 'stop_lon' which contain the latitude
+        and longitude values.
+
+    Returns
+    -------
+    pd.DataFrame
+        `stops` with additional 'stop_east' and 'stop_north' columns
+        appended, containing the easting and northing values. Any
+        translation errors will cause the new columns to contain NaNs.
+    """
+    eastnorth = pd.DataFrame(
```

Comment:
```suggestion
    required_cols = ["stops_lat", "stops_lon"]
    missing_cols = [col for col in required_cols if col not in stops.columns]
    if missing_cols:
        raise ValueError(f"`stops` DataFrame is missing the required columns: {missing_cols}")
    
    eastnorth = pd.DataFrame(
```

Reply:
Done based on your suggestion in f39e0bb6da74409e35fb6e89593941fb07d76fd0